FlightScheduler/APICall.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  APICall.swift
+//  FlightScheduler
+//
+//  Created by macbook on 26/02/2020.
+//  Copyright © 2020 Practice. All rights reserved.
+//
+
+import Foundation
+
+enum APIError: Error {
+    case responseProblem
+    case decodingProblem
+    case encodingProblem
+
+}
+
+let GRANT_TYPE = "credentials"
+let CLIENT_ID = "xuzza9nuseytv3m59fyytrue"
+let CLIENT_SECRET = "mCH385mArn"
+
+struct APIRequest {
+    let resourceURL: URL
+
+    init(endpoint: String) {
+        let resourceString = "https://api.lufthansa.com/v1/oauth/token/"
+        guard let resourceURL = URL(string: resourceString) else { fatalError() }
```

Comment:
It's not good practise to do something like this in the initialiser. Just do the initialiasing in the init